src/lib/components/Swap/Output.tsx

```diff
@@ -33,32 +36,34 @@ interface OutputProps {
 }
 
 export default function Output({ disabled, children }: OutputProps) {
-  const input = useAtomValue(inputAtom)
-  const output = useAtomValue(outputAtom)
-  const setValue = useUpdateInputValue(outputAtom)
-  const setToken = useUpdateInputToken(outputAtom)
-  const balance = 123.45
+  const { onCurrencySelection, onUserInput } = useSwapActionHandlers()
+  const {
+    currencies: { [Field.INPUT]: inputCurrency, [Field.OUTPUT]: outputCurrency },
+    parsedAmounts: { [Field.INPUT]: inputAmount, [Field.OUTPUT]: outputAmount },
+    currencyBalances: { [Field.OUTPUT]: balance },
+  } = useDerivedSwapInfo()
 
   const overrideColor = useAtomValue(colorAtom)
-  const dynamicColor = useCurrencyColor(output.token)
-  usePrefetchCurrencyColor(input.token) // extract eagerly in case of reversal
+  const dynamicColor = useCurrencyColor(outputCurrency)
+  usePrefetchCurrencyColor(inputCurrency) // extract eagerly in case of reversal
   const color = overrideColor || dynamicColor
-  const hasColor = output.token ? Boolean(color) || null : false
+  const hasColor = outputCurrency ? Boolean(color) || null : false
 
+  const inputUSDCValue = useUSDCValue(inputAmount)
```

Comment:
nit: This calls it `inputUSDCValue`, but `Input.tsx` calls it `inputAmountUSDC`. Rename both to `inputUSDC`?